subsys/net/lib/tftp/tftp_client.c

```diff
@@ -0,0 +1,344 @@
+	 * is created per RFC1350.
+	 */
+
+	/* Fill in the Request Type. Also keep tabs on the request size. */
+	put_u16(tftpc_buffer, request_type);
+	req_size = 2;
+
+	/* Copy in the name of the remote file - the file we need to get
+	 * from the server. Add an upper bound to ensure no buffers overflow.
+	 */
+	strncpy(tftpc_buffer + req_size, remote_file, TFTP_MAX_FILENAME_SIZE);
+	req_size += strlen(remote_file);
+
+	/* Fill in 0. */
+	tftpc_buffer[req_size] = 0x0;
+	req_size++;
+
+	/* Default to "Octet" if mode not specified. */
+	if (mode == NULL)
+		mode = def_mode;
+
+	/* Copy the mode of operation. */
+	strncpy(tftpc_buffer + req_size, mode, TFTP_MAX_MODE_SIZE);
+	req_size += strlen(mode);
+
+	/* Fill in 0. */
+	tftpc_buffer[req_size] = 0x0;
+	req_size++;
+
+	return req_size;
+}
+
+/* Name: send_err
+ * Description: This function sends an Error report to the TFTP Server.
+ */
+static inline int send_err(int sock, int err_code, char *err_string)
+{
+	u32_t req_size;
+
+	LOG_ERR("Client Error. Sending code: %d(%s)", err_code, err_string);
+
+	/* Fill in the "Err" Opcode and the actual error code. */
+	put_u16(tftpc_buffer, ERROR_OPCODE);
+	put_u16(tftpc_buffer + 2, err_code);
+	req_size = 4;
+
+	/* Copy the Error String. */
+	strcpy(tftpc_buffer + req_size, err_string);
+	req_size += strlen(err_string);
+
+	/* Send Error to server. */
+	return send(sock, tftpc_buffer, req_size, 0);
+}
+
+/* Name: tftpc_recv_data
+ * Description: This function tries to get data from the TFTP Server
+ * (either response or data). Times out eventually.
+ */
+static int tftpc_recv_data(int sock)
+{
+	int     stat;
+
+	/* Enable select on the socket. */
+	pollfds_add(sock);
+
+	/* Enable select on this socket. Wait for the specified number
+	 * of seconds before exiting the call.
+	 */
+	stat = select(fdnum, &readfds, NULL, NULL, &tftpc_timeout);
+
+	/* Did we get some data? Or did we time out? */
+	if (stat > 0) {
+
+		/* Receive data from the TFTP Server. */
+		stat = recv(sock, tftpc_buffer, TFTPC_MAX_BUF_SIZE, 0);
+
+		/* Store the amount of data received in the global variable. */
+		tftpc_buffer_size = stat;
+	}
+
+	return stat;
+}
+
+/* Name: tftpc_process_resp
+ * Description: This function will process the data received from the
+ * TFTP Server (a file or part of the file) and place it in the user buffer.
+ * Return Value: This function will return one of the following values,
+ * -> TFTPC_DUPLICATE_DATA: If the block is already received by the client.
+ * -> TFTPC_BUFFER_OVERFLOW: If the data is more than the user provided buffer.
+ * -> TFTPC_DATA_RX_SUCCESS: data received but their is more to come.
+ * -> TFTPC_SUCCESS: Block received successfully and no more data is coming.
+ */
+static int tftpc_process_resp(int sock, struct tftpc *client)
+{
+	u16_t    block_no;
+
+	/* Get the block number as received in the packet. */
+	block_no = get_u16(tftpc_buffer + 2);
+
+	if (tftpc_block_no > block_no) {
+
+		LOG_INF("Duplicate block received: %d", block_no);
```

Comment:
I see lot of `LOG_INF()` messages printed in this code. Are you sure you want that, usually the INFO level is not very useful, and probably debug would be more appropriate.

Reply:
Made all changes including this.. Let me know what more is required.. I will also be pushing a sample application once this PR is merged..